src/AutoMapper/XpressionMapper/XpressionMapperVisitor.cs

```diff
@@ -254,7 +254,7 @@ protected void FindDestinationFullName(Type typeSource, Type typeDestination, st
                 var propertyName = sourceFullName.Substring(0, sourceFullName.IndexOf(period, StringComparison.OrdinalIgnoreCase));
                 var propertyMap = typeMap.GetPropertyMaps().SingleOrDefault(item => item.DestinationProperty.Name == propertyName);
                 
-                var sourceMemberInfo = typeSource.GetAllMembers().First(mi => mi.Name == propertyMap.DestinationProperty.Name);
+                var sourceMemberInfo = typeSource.GetInheritedMember(propertyMap.DestinationProperty.Name);

```

Comment:
This might be confusing, as the original "TypeHelper.GetAllMembers()" function also returns inherited members.
What I really changed is, that it also returned interface members
So maybe it should be called `GetClassAndInterfaceMembers` ?